ring/partition_instance_ring.go

```diff
@@ -0,0 +1,123 @@
+package ring
+
+import (
+	"fmt"
+	"time"
+)
+
+type PartitionRingReader interface {
+	PartitionRing() *PartitionRing
```

Comment:
Let's mention that we never expect to get `nil` from this method.

Reply:
Agree. Done in https://github.com/grafana/dskit/pull/479/commits/48ffab9b07a12f8f692da0fc5d47404c510712ff